iceoryx_posh/source/roudi/roudi.cpp

```diff
@@ -578,5 +585,103 @@ void RouDi::IpcMessageErrorHandler() noexcept
 {
 }
 
+namespace service_management
+{
+
+ServiceManagementSystemd::~ServiceManagementSystemd()
+{
+    if (m_listenThreadWatchdog.joinable())
+    {
+        m_listenThreadWatchdog.join();
+    }
+}
+
+void ServiceManagementSystemd::shutdown()
+{
+    m_shutdown.store(true);
+}
+
+std::string ServiceManagementSystemd::getEnvironmentVariable(const char* const env_var)
+{
+    iox::string<SIZE_STRING> str;
+
+    str.unsafe_raw_access([&](auto* buffer, auto const info) {
+        size_t actualSizeWithNull{0};
+        auto result =
+            IOX_POSIX_CALL(iox_getenv_s)(&actualSizeWithNull, buffer, static_cast<size_t>(info.total_size), env_var)
+                .failureReturnValue(-1)
+                .evaluate();
+
+        if (result.has_error() && result.error().errnum == ERANGE)
+        {
+            IOX_LOG(ERROR, "Invalid value for '" + std::string(env_var) + "' environment variable!");
+        }
+
+        size_t actual_size{0};
+        constexpr size_t NULL_TERMINATOR_SIZE{1};
+        if (actualSizeWithNull > 0)
+        {
+            actual_size = actualSizeWithNull - NULL_TERMINATOR_SIZE;
+        }
+        buffer[actual_size] = 0;
+        return actual_size;
+    });
+
+    return std::string(str.c_str());
+}
+
+bool ServiceManagementSystemd::setThreadNameHelper(iox::string<SIZE_THREAD_NAME>& threadName)
+{
+    bool status_change_name = iox::setThreadName(threadName);
+    if (!status_change_name)
+    {
+        IOX_LOG(ERROR, "Cannot set name for thread " << threadName);
+        return false;
+    }
+    return true;
+}
+
+void ServiceManagementSystemd::watchdogLoopHelper()
+{
+    IOX_LOG(INFO, "Start watchdog");
+    while (!m_shutdown.load())
+    {
+        std::unique_lock<std::mutex> lock(watchdogMutex);
+        if (watchdogNotifyCondition.wait_for(lock, std::chrono::seconds(1), [this] { return m_shutdown.load(); }))
+        {
+            break;
+        }
+        if (!sendSDNotifySignalHelper("WATCHDOG=1"))
+        {
+            return;
+        }
+    }
+}
+
+void ServiceManagementSystemd::processNotify()
```

Comment:
Also just nitpicking. For someone not that familiar with systemd, having this method called `notifyProcessReady` gives a better understanding what it does.